File: 2017/ViewAnimations/AnimationResources/app/src/main/java/me/mehdi/animationresources/MainActivity.java
package me.mehdi.animationresources;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.view.animation.Animation;
import android.view.animation.AnimationUtils;
import android.widget.Button;
import android.widget.ImageView;

public class MainActivity extends AppCompatActivity {

    Button startButton;
    ImageView heartImage;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        startButton = findViewById(R.id.start);
        heartImage = findViewById(R.id.heart);

        startButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Animation animation1 = AnimationUtils.loadAnimation(MainActivity.this, R.anim.animate_one);
                Animation animation2 = AnimationUtils.loadAnimation(MainActivity.this, R.anim.animate_two);

                v.startAnimation(animation1);
                heartImage.startAnimation(animation2);
            }
        });

    }
}

File: ViewAnimation/app/src/main/java/com/example/viewanimation/MainActivity.kt
package com.example.viewanimation

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.animation.*
import com.example.viewanimation.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {
    private lateinit var binding: ActivityMainBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)



        binding = ActivityMainBinding.inflate(layoutInflater)


        setContentView(binding.root)


        title = "MainActivity"
        binding.start.setOnClickListener {button->
            val myResAnimSet = AnimationUtils.loadAnimation(this, R.anim.my_anim)

            button.startAnimation(myResAnimSet)
//            val myAnimationSet = AnimationSet(true)
//            myAnimationSet.duration = 3000
//            val translate = TranslateAnimation(
//                0f,
//            300f,
//                 0f,
//                 300f
//            )
//
//            val rotate = RotateAnimation(0f, 360f)
//            val alpha = AlphaAnimation(1f, 0f)
//            val scale = ScaleAnimation(
//                1f, 3f,
//            1f, 3f)
//
//
//            myAnimationSet.apply{
//                addAnimation(translate)
//                addAnimation(rotate)
//                addAnimation(alpha)
//                addAnimation(scale)
//            }
//
//            button.startAnimation(myAnimationSet)

        }


        binding.startActivity.setOnClickListener {
            val intent = Intent(this, SecondActivity::class.java)
            startActivity(intent)
            overridePendingTransition(R.anim.slide_right_in, R.anim.slide_left_out)

        }


    }
}